src/test/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManagerTest.java

```diff
@@ -89,4 +168,319 @@ private static Stream<Arguments> emailsToTest() {
         Arguments.of(false, "testemail-twittter")
     );
   }
+
+  @Test
+  public void updateTeacherPendingFlag_success() throws SegueDatabaseException, NoUserException {
+    RegisteredUser initialUserState = new RegisteredUser() {
```

Comment:
Very interesting, I had never seen this pattern before. I have been reading a bit about it, and this is what I found, though:

> Disadvantages of using double braces are:
> - Obscure, not widely known way to do the initialization
> - It creates an extra class every time we use it
> - Doesn’t support the use of the “diamond operator” – a feature introduced in Java 7
> - Doesn’t work if the class we are trying to extend is marked final
> - Holds a hidden reference to the enclosing instance, which may cause memory leaks
>
> It’s due to these disadvantages that double brace initialization is considered as an anti-pattern.

Reply:
Ah, I'd seen it used in some of the existing tests and didn't realise it had so many issues...
Should I see about switching these over to a different method then?